pkg/shared/util/sasl_config.go

```diff
@@ -37,6 +40,23 @@ func GetSASL(saslConfig *dfv1.SASL) (*struct {
 	SCRAMClientGeneratorFunc func() sarama.SCRAMClient
 	TokenProvider            sarama.AccessTokenProvider
 	GSSAPI                   sarama.GSSAPIConfig
+}, error) {
+	return GetSASLStrategy(saslConfig, OsFile{})
+}
+
+// GetSASLConfig A utility function to get sarama.Config.Net.SASL
+func GetSASLStrategy[GetSecretsStrategy GetSecrets](saslConfig *dfv1.SASL, strategy GetSecretsStrategy) (*struct {
```

Comment:
What is the reason you use generics instead of just writing it as:

```
func GetSASLStrategy(saslConfig *dfv1.SASL, strategy GetSecrets) (*struct {
...
```

Reply:
Also, make the function private.

Reply:
>What is the reason you use generics instead of just writing it as:
That's a very good point! Changed!

> Also, make the function private.
I went through the code I changed the visibility of things that should be private